### DISCUSSION TITLE
Next.js installation instructions out of date?

### DISCUSSION THREAD

**Discussion by CharlesWiltgen** [tailwindlabs/tailwindcss#3420] [Help] [answered]
_(I'm new to both Tailwind and Next, so I hope this isn't too trivial an issue to post about.)_

At [https://tailwindcss.com/docs/guides/nextjs](https://tailwindcss.com/docs/guides/nextjs), it says:

> If you aren't planning to write any custom CSS in your project, the fastest way to include Tailwind is to import it directly in `pages/_app.js`:

There is no `pages/_app.js` in a fresh installation of the current version of Next.js, so I assume these instructions were for a previous version. There is an `index.js`, but it looks nothing like the contents of `_app.js` mentioned in the Tailwind installation instructions.

> Open the ./styles/globals.css file that Next.js generates for you by default…

This file is (apparently) no longer generated by default. My guess would be that I need to create a `globals.css` in `static`, but since I'm so new to both I'd appreciate a best-practices recommendation for this as well.

(It's very possible subsequent instructions are incorrect, but I stopped there for now. Thank you!)

**Comment by DorisKnight** [comment]
Next. js provides a backend that can server side render a response to request, allowing you to create a dynamic website, which means you will deploy it on a platform that can run Node.
[MyPrepaidCenter - Login to Check Balance or Activate Card](https://www.myprepaidcenter.us/)

**Reply by CharlesWiltgen** [reply]
> Next. js provides a backend that can server side render a response to request…

I appreciate the response @DorisKnight, but I don't have a question about deploying Node.js apps. (I've done that several times.)

TLDR: The **Tailwind setup instructions for Node.js 10 are either wrong or incomplete**. I'd appreciate guidance on the specific questions above, but now I'm additionally wondering if I should just file this as a documentation bug instead.

**Comment by adamwathan** [comment, marked as answer]
Hey not sure how you are installing Next.js, but this is the directory structure for a fresh Next app:

![image](https://user-images.githubusercontent.com/4323180/105614093-0f1cfb80-5d95-11eb-9793-42ebc613b9f0.png)

That's what you get when running these instructions from the Next.js docs:

![image](https://user-images.githubusercontent.com/4323180/105614125-3d9ad680-5d95-11eb-917a-f392544d6686.png)

The instructions in the docs are accurate, there is no `static` folder in Next.js 10 (there used to be in much older versions) so maybe you are somehow working with an old version of next.

**Reply by CharlesWiltgen** [reply]
Hey, @adamwathan! Thanks for the response. You're the reason I'm making the leap to Tailwind. 🙂 For better or worse, I'm learning Next.js at the same time.

`npx next -v` says I'm using Next.js v10.0.5, and the directory structure created by `npx create-next-app test` doesn't look like yours. Here's what I get:

<img width="204" alt="image" src="https://user-images.githubusercontent.com/6807178/105615303-69619080-5d84-11eb-9d22-ac78be01805c.png">

That explains why the Tailwind instructions don't work for me, since there's (for example) no `_app.js` or even an `api` directory. Any thoughts on why our results are different?

**Reply by adamwathan** [reply]
What's in your package.json? Next deprecated the static folder in favor of public quite a while ago.

**Reply by adamwathan** [reply]
Also can you show me the exact commands you typed to create your project?

**Reply by CharlesWiltgen** [reply]
> What's in your package.json?

`npx create-next-app test` creates a package.json containing this:

```
{
  "name": "create-next-app-default",
  "version": "0.1.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "next start"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "next": "^10.0.5",
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  }
}
```

> Also can you show me the exact commands you typed to create your project?

Sure, here's the command I entered followed by the output:

```
npx create-next-app test

Creating test...

> Success! Created files for "test" next app

  Installing npm modules:
    react
    react-dom
    next

> Success! Installed dependencies for test
```

**Reply by adamwathan** [reply]
Do you get the same result using this?

`npx create-next-app@latest test`

Could be that you have an old cached version of create-next-app.

**Reply by CharlesWiltgen** [reply]
> Could be that you have an old cached version of create-next-app.

That solved it—thank you, @adamwathan! 🙌 I really appreciate the help.

(In case it's technically interesting, `create-next-app` didn't (and doesn't appear) when I do a `npm list -g create-next-app`.)
